Tools/Scripts/webkitpy/layout_tests/controllers/layout_test_runner.py

```diff
@@ -504,6 +505,70 @@ def __init__(self, name, test_inputs):
         self.test_inputs = test_inputs
         self.needs_servers = test_inputs[0].needs_servers
 
+    def shorten(self, string):
+        if not string:
+            return string
+        return string.replace('{}/'.format(self.name), '{}')
+
+    def expand(self, string):
+        if not string:
+            return string
+        return string.replace('{}', '{}/'.format(self.name))
+
+    def pack(self, test_input):
+        return TestInput(
+            test=Test(
+                test_path=self.shorten(test_input.test.test_path),
+                expected_text_path=self.shorten(test_input.test.expected_text_path),
+                expected_image_path=self.shorten(test_input.test.expected_image_path),
+                expected_checksum_path=self.shorten(test_input.test.expected_checksum_path),
+                expected_audio_path=self.shorten(test_input.test.expected_audio_path),
+                reference_files=None if test_input.test.reference_files is None else [self.shorten(file) for file in test_input.test.reference_files],
+                is_http_test=test_input.test.is_http_test,
+                is_websocket_test=test_input.test.is_websocket_test,
+                is_wpt_test=test_input.test.is_wpt_test,
+                is_wpt_crash_test=test_input.test.is_wpt_crash_test,
+            ), timeout=test_input.timeout,
+            is_slow=test_input.is_slow,
+            needs_servers=test_input.needs_servers,
+            should_dump_jsconsolelog_in_stderr=test_input.should_dump_jsconsolelog_in_stderr,
+            reference_files=test_input.reference_files,
+            should_run_pixel_test=test_input.should_run_pixel_test,
+        )
+
+    def unpack(self, test_input):
```

Comment:
It looks like `pack` and `unpack` are almost identical, the only difference is how we operated on path, maybe we can have a helper function to take `self.shorten` and `self.expand` as argument.